File: Project1/src/Lion.java
import java.util.ArrayList;
import java.util.Random;

public class Lion extends Feline{

    final String[] lionSounds = {"meow", "roar", "grunt", "moan", "growl", "snarl", "purr", "hum", "puff", "woof"};

    Lion() {
        super(false, "Carnivore", "Grassland", true, "Lenny", 4);
    }

    /**
     *  Constructor with only Animal parameters
     * @param awake:boolean - Is the animal awake?
     * @param diet:String - What does the animal eat?
     * @param habitat:String - Where does the animal live?
     * @param hungry:boolean - Is the animal hungry?
     * @param name:String - The name of the animal
     * @param numLegs:int - How many legs does the animal have?
     */
    Lion(boolean awake, String diet, String habitat, boolean hungry, String name, int numLegs){
        super(awake, diet, habitat, hungry, name, numLegs);
    }

    /**
     * Override for makeNoise function
     * Constructs a random string of 5 lion sounds
     */
    @Override
    public void makeNoise() {
        Random rd = new Random();
        String noise = "";
        for(int i = 0; i < 5; i++){
            String nextSound = lionSounds[rd.nextInt(lionSounds.length)] + " ";
            noise += nextSound;
        }
        System.out.println(name + " the " + this.getClass().getName() + ": Made the following noises - " + noise);
    }


}

File: Project2/Lion.py
import random
from Feline import Feline

class Lion(Feline):

  '''
    Generic initializer for Lion
    @param name:String, default:'Lily'
    @param awake:Boolean, default:False
    @param hungry:Boolean, deault:True
    @param num_legs:Int, default:4
  '''
  def __init__(self, name='Lily', awake=False, hungry=True, num_legs=4):
    super(Lion, self).__init__(name, awake, hungry, num_legs)

  '''
    Lion makes a noise with the following probability
      - 25% chance to purr
      - 25% chance to grumble
      - 25% chance to roar loudly
  '''
  def make_noise(self):
    animal_type = type(self).__name__
    random_number = random.random()
    if(random_number < 0.25):
      print('{} the {} purred.'.format(self.name, animal_type))
    elif(random_number < 0.5):
      print('{} the {} grumbles.'.format(self.name, animal_type))
    else:
        print('{} the {} roars loudly.'.format(self.name, animal_type))
